src/modules/did/did.service.ts

```diff
@@ -358,10 +363,30 @@ export class DIDService implements OnModuleInit, OnModuleDestroy {
 
   private async syncDocuments() {
     this.logger.debug(`Beginning sync of DID Documents`);
-    const cachedDIDs = await this.didRepository.find({ select: ['id'] });
-    cachedDIDs.forEach(async (did) => {
+    const syncs = await this.latestDidSyncRepository.find({
+      order: { createdDate: 'DESC' },
+    });
+    const fromBlock = syncs.length > 0 ? syncs[0].block : 0;
+    const topBlock = await this.provider.getBlockNumber();
+    this.logger.debug(
+      `Getting DID update events from block ${fromBlock} to block ${topBlock}...`
+    );
+    const changedIdentities = await this.getChangedIdentities(
+      fromBlock,
+      topBlock
+    );
+    const didsToSynchronize = (
+      await this.didRepository.find({ select: ['id'] })
+    ).filter((doc) => {
+      const identity = doc.id.split(':')[3];
```

Comment:
I think there might be a helpful function for this in one of our libraries. Something like `getAddress` perhaps?